ddtrace/opentelemetry/span.go

```diff
@@ -68,8 +69,13 @@ func (s *span) SpanContext() oteltrace.SpanContext {
 	ctx := s.Span.Context()
 	var traceID oteltrace.TraceID
 	var spanID oteltrace.SpanID
-	// todo(dianashevchenko): change ctx.TraceID() to extract 128 traceID from W3C interface
-	uint64ToByte(ctx.TraceID(), traceID[:])
+	w3cCtx, ok := ctx.(ddtrace.SpanContextW3C)
+	if !ok {
+		log.Debug("Non-W3C context found in span, unable to get full 128 bit trace id")
```

Comment:
We don't expect this to happen often, in theory shouldn't happen at all. It might be the case if mixed tracerProviders used, but I would expect this to basically never happen